src/main/java/org/folio/circulation/domain/notice/schedule/FeeFineScheduledNoticeHandler.java

```diff
@@ -0,0 +1,134 @@
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.Result;
+import org.joda.time.DateTime;
+import org.joda.time.Period;
+
+import io.vertx.core.json.JsonObject;
+
+public class FeeFineScheduledNoticeHandler {
+  private final PatronNoticeService patronNoticeService;
+  private final ScheduledNoticesRepository scheduledNoticesRepository;
+  private final FeeFineActionRepository actionRepository;
+  private final AccountRepository accountRepository;
+
+  private FeeFineScheduledNoticeHandler(PatronNoticeService patronNoticeService,
+    ScheduledNoticesRepository scheduledNoticesRepository,
+    FeeFineActionRepository actionRepository,
+    AccountRepository accountRepository) {
+
+    this.patronNoticeService = patronNoticeService;
+    this.scheduledNoticesRepository = scheduledNoticesRepository;
+    this.actionRepository = actionRepository;
+    this.accountRepository = accountRepository;
+  }
+
+  public static FeeFineScheduledNoticeHandler using(Clients clients) {
+    return new FeeFineScheduledNoticeHandler(
+      PatronNoticeService.using(clients),
+      ScheduledNoticesRepository.using(clients),
+      new FeeFineActionRepository(clients),
+      new AccountRepository(clients));
+  }
+
+  public CompletableFuture<Result<List<ScheduledNotice>>> handleNotices(
+    Collection<ScheduledNotice> scheduledNotices) {
+
+    return allOf(scheduledNotices, this::handleNotice);
+  }
+
+  public CompletableFuture<Result<ScheduledNotice>> handleNotice(ScheduledNotice notice) {
+    return ofAsync(FeeFineRelatedRecords::new)
+      .thenCompose(r -> r.combineAfter(records ->
+        actionRepository.findById(notice.getFeeFineActionId()),
+        FeeFineRelatedRecords::withAction))
+      .thenCompose(r -> r.combineAfter(this::fetchAccount, FeeFineRelatedRecords::withAccount))
+      .thenCompose(r -> r.after(records -> processNotice(notice, records)));
+  }
+
+  private CompletableFuture<Result<Account>> fetchAccount(FeeFineRelatedRecords records) {
+    FeeFineAction action = records.getAction();
+
+    return action != null
+      ? accountRepository.findById(action.getAccountId())
+      : ofAsync(() -> null);
+  }
+
+  private CompletableFuture<Result<ScheduledNotice>> processNotice(
+    ScheduledNotice notice, FeeFineRelatedRecords records) {
+
+    if (records.isValid()) {
+      FeeFineAction action = records.getAction();
+      JsonObject noticeContext = createFeeFineNoticeContext(records.getAccount(), action);
+      ScheduledNoticeConfig config = notice.getConfiguration();
+
+      patronNoticeService.acceptScheduledNoticeEvent(config, action.getUserId(), noticeContext);
+
+      if (config.isRecurring()) {
+        return scheduledNoticesRepository.update(getNextRecurringNotice(notice));
+      }
+    }
+
+    return scheduledNoticesRepository.delete(notice);
+  }
+
+  private ScheduledNotice getNextRecurringNotice(ScheduledNotice notice) {
```

Comment:
To me personally "get" in the method name means that we either getting object's property or getting some other object by certain criteria. In this case it looks like we're updating notice's next run time property. Would something like updateNoticeNextRunTime() be more appropriate?

Reply:
Sure, done.